packages/components/src/Heading/Heading.tsx

```diff
@@ -55,21 +55,19 @@ const HeadingComponent = styled.span<BaseHeadingProps>(
   ]
 );
 
-function Heading(props: Props) {
-  const {
-    as,
-    bold = true,
-    children,
-    size,
-    /**
-     * Components that wrap typography sometimes require props such as
-     * event handlers, tabIndex, ref, etc. to be passed down into the element.
-     * TODO: add better typing or documentation for optional props,
-     * e.g. React.HTMLAttributes<HTMLHeadingElement>
-     */
-    ...rest
-  } = props;
-
+function Heading({
+  as,
+  bold = true,
+  children,
+  size,
+  /**
+   * Components that wrap typography sometimes require props such as
+   * event handlers, tabIndex, ref, etc. to be passed down into the element.
```

Comment:
We may want to remove reference to `ref` here, since I don't think it would work (without `forwardRef`)